platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/AttributeSharingSessionDataStore.java

```diff
@@ -49,17 +49,14 @@
    * @param sessionAttributes the session's attributes
    */
   public void updateSessionAttributes(String id, Map<String, Object> sessionAttributes) {
-    SessionData sessionData;
     synchronized (sessionDataMap) {
-      sessionData = sessionDataMap.get(id);
-    }
+      SessionData sessionData = sessionDataMap.get(id);
 
-    if (sessionData != null && !sessionData.getAllAttributes().equals(sessionAttributes)) {
-      LOGGER.trace(
-          "Storing new attributes for session {} at context {}",
-          id,
-          _context.getCanonicalContextPath());
-      synchronized (sessionDataMap) {
+      if (sessionData != null && !sessionData.getAllAttributes().equals(sessionAttributes)) {
+        LOGGER.trace(
+            "Storing new attributes for session {} at context {}",
+            id,
+            _context.getCanonicalContextPath());
```

Comment:
`newSessionData() below ... should it not cache the session being created?

Reply:
It hasn't created a new Map. Rather, it's just clearing and putting new elements in the already cached Map.

Reply:
I misunderstood the question in the above response, but @paouelle and I hashed it out offline. The `Map` created from `newSessionData` is later cached when Jetty's `SessionHandler` calls the `doStore` method.